websites/B/bilibilitv/presence.ts

```diff
@@ -1,97 +1,164 @@
 const presence = new Presence({
-		clientId: "543380687466528772"
+		clientId: "972073369564483584"
 	}),
-	browsingStamp = Math.floor(Date.now() / 1000),
-	urlpath = document.location.pathname.split("/"),
-	getElement = (query: string): string | undefined => {
-		return document.querySelector(query)?.textContent;
-	},
-	getEpisode = (query: string): string | undefined => {
-		return document
-			.querySelector(query)
-			?.textContent.split("E")[1]
-			.replace(/- Bstation/gi, "");
-	};
+	browsingStamp = Math.floor(Date.now() / 1000);
 
-presence.on("UpdateData", async () => {
-	const presenceData: PresenceData = {
-			largeImageKey: "icon",
-			details: "Idle"
+async function getStrings() {
+	return presence.getStrings(
+		{
+			play: "general.playing",
+			pause: "general.paused",
+			search: "general.search",
+			episode: "general.episode",
+			browsing: "general.browsing",
+			viewHome: "general.viewHome",
+			viewChannel: "general.viewChannel",
+			watchingVid: "general.watchingVid",
+			viewCategory: "general.viewCategory",
+			readingAbout: "general.readingAbout",
+			searchSomething: "general.searchSomething",
+			buttonWatchVideo: "general.buttonWatchVideo",
+			buttonViewEpisode: "general.buttonViewEpisode"
 		},
-		path = document.location.pathname.toLowerCase();
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
 
+presence.on("UpdateData", async () => {
+	const [newLang] = await Promise.all([
+			presence.getSetting<string>("lang").catch(() => "en")
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: "main",
+			startTimestamp: browsingStamp
+		},
+		title =
+			document.querySelector(
+				"div.layout-body.media-width > div > ul > li:nth-child(3) > span"
+			)?.textContent ||
+			document.querySelector(
+				"div.layout-normal > div.layout-body.media-width > div > ul > li:nth-child(2) > a"
+			)?.textContent,
+		playing = document.querySelector(
+			"img.player-mobile-icon.player-mobile-pause-icon.player-mobile-active"
+		);
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
 	// Main Site
-
-	if (path === "/en") {
-		presenceData.smallImageKey = "idle";
-		presenceData.details = "Currently browsing";
-		presenceData.state = "Homepage";
-	} else if (path.startsWith("/en/play")) {
-		presenceData.smallImageKey = "play";
-		presenceData.smallImageText = "Watching a show";
-		presenceData.details = getElement("h1.video-info__title");
-		presenceData.state = `Episode ${getEpisode("title")}`;
-		presenceData.startTimestamp = browsingStamp;
-		presenceData.buttons = [
-			{
-				label: "Watch this show",
-				url: `https://www.bilibili.tv/en/play/${urlpath[3]}`
+	if (document.location.hostname === "www.bilibili.tv") {
+		switch (document.location.pathname.toString().toLowerCase().split("/")[2]) {
```

Comment:
you're stringifying a string?
```suggestion
		switch (document.location.pathname.toLowerCase().split("/")[2]) {
```